### PR TITLE
Move the core CLI related docs to cli/core/docs folder

### PR DESCRIPTION
**Pull request by prkalle**
Signed-off-by: Prem Kumar Kalle <pkalle@vmware.com>

### What this PR does / why we need it
This PR moves the core CLI related docs to the  refactored CLI folder docs( cli/core/docs)

### Which issue(s) this PR fixes

Fixes #related to 3301 

### Release note

```release-note
N/A
```

### CODE REVIEW

File: cli/core/docs/design/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Design Proposals
+
+This directory is to be populated with design reviews.
+
+Use the [Proposal Template](docs/cli/dev/_proposal.md) as a starting point.
```
Comment (marckhouzam) [inline cli/core/docs/design/README.md:5]: We should probably copy that file into `cli/core` as well to keep this link working

File: cli/core/docs/cli/style_guide.md [hunk 2]
```diff
@@ -193,7 +193,7 @@ $ tanzu apps workload delete <name> --wait-timeout 10m
 
 Available for plugins written in golang.
 
-CLI commands should utilize the plugin component library in [pkg/cli/component](https://github.com/vmware-tanzu/tanzu-framework/tree/main/pkg/v1/cli/component) for interactive features like prompts or table printing.
+CLI commands should utilize the plugin component library in [cli/runtime/component](https://github.com/vmware-tanzu/tanzu-framework/tree/main/cli/runtime/component) for interactive features like prompts or table printing.
```
Comment (marckhouzam) [inline cli/core/docs/cli/style_guide.md:196]: Nice job fixing the links 👍 

I found another one at line 134 that needs ajusting

File: cli/core/docs/cli/plugin_implementation_guide.md [hunk 3]
```diff
@@ -5,7 +5,7 @@
 The Tanzu CLI was built to be extensible across teams and be cohesive across SKUs. To this end, the Tanzu CLI provides
```
Comment (marckhouzam) [inline cli/core/docs/cli/plugin_implementation_guide.md:5]: Also need to fix links at lines: 
- 42
- 162
-

File: cli/core/docs/cli/cli-architecture.md [hunk 4]
```diff
@@ -159,7 +159,7 @@ tanzu mission-control cluster list
 
 ## Plugin Discovery Sources
 
-NOTE: This is applicable only if [context-aware plugin discovery](/docs/design/context-aware-plugin-discovery-design.md) is enabled within Tanzu CLI.
+NOTE: This is applicable only if [context-aware plugin discovery](../design/context-aware-plugin-discovery-design.md) is enabled within Tanzu CLI.
 
 Discovery is the interface to fetch the list of available plugins, their supported versions and how to download them either standalone or scoped to a context(server). E.g., the CLIPlugin API in a management cluster, OCI based plugin discovery for standalone plugins, a similar REST API and a manifest file in GCP based discovery, etc. (API is defined [here](apis/config/v1alpha1/clientconfig_types.go#L111-L187)) Unsupported plugins and plugin versions are not returned by the interface. Having a separate interface for discovery helps to decouple discovery (which is usually tied to a server or user identity) from distribution (which can be shared).
```
Comment (marckhouzam) [inline cli/core/docs/cli/cli-architecture.md:164]: The link here is broken I believe.

**Review by marckhouzam — COMMENTED** [review]
1. I think we should grep for all the name of files we are moving and update the links that are broken.  For example, the following is broken: https://github.com/vmware-tanzu/tanzu-framework/blob/c9c55552df51b9187ad64121f0a17d3db7c50fd2/docs/framework-use-cases.md?plain=1#L21
1. We can probably remove `docs/design/images/context-aware-plugin-discovery-design.png` since we moved it and there are no other references to it.

File: cli/core/docs/cli/plugin_implementation_guide.md (hunk 3 above)
Comment (prkalle) [reply cli/core/docs/cli/plugin_implementation_guide.md:5]: Fixed.

File: cli/core/docs/cli/cli-architecture.md (hunk 4 above)
Comment (prkalle) [reply cli/core/docs/cli/cli-architecture.md]: Fixed. Thanks!

File: cli/core/docs/cli/style_guide.md (hunk 2 above)
Comment (prkalle) [reply cli/core/docs/cli/style_guide.md:196]: Thank you. Fixed.

File: cli/core/docs/design/README.md (hunk 1 above)
Comment (prkalle) [reply cli/core/docs/design/README.md:5]: Updated the link. Current plan is to still share the `dev` under `cli/core/docs` as it has docs like `build.md` which are still valid for cli/core.

**Review by marckhouzam — APPROVED** [review]
Thanks @prkalle for addressing all the comments.

LGTM

File: cli/core/docs/cli/cli-architecture.md [hunk 5]
```diff
@@ -159,9 +159,9 @@ tanzu mission-control cluster list
...
+Discovery is the interface to fetch the list of available plugins, their supported versions and how to download them either standalone or scoped to a context(server). E.g., the CLIPlugin API in a management cluster, OCI based plugin discovery for standalone plugins, a similar REST API and a manifest file in GCP based discovery, etc. (API is defined [here](../../../../apis/config/v1alpha1/clientconfig_types.go)) Unsupported plugins and plugin versions are not returned by the interface. Having a separate interface for discovery helps to decouple discovery (which is usually tied to a server or user identity) from distribution (which can be shared).
```
Comment (anujc25) [inline cli/core/docs/cli/cli-architecture.md:164]: nit: I might need to update `../../../../apis/config/v1alpha1/clientconfig_types.go` path as part of PR #3375 👍

File: cli/core/docs/cli/plugin_implementation_guide.md [hunk 6]
```diff
@@ -39,7 +39,7 @@ NOTE: We are working on enhancing this user experience by publishing admin artif
 
 Context is an isolated scope of relevant client-side configurations for a combination of user identity and server identity. There can be multiple contexts for the same combination of `(user, server)`. Previously, this was referred to as `Server` in the Tanzu CLI. Going forward we shall refer to them as `Context` to be explicit.
 
-If a plugin wants to access the context it should use the [provided libraries](/pkg/v1/config/context.go) for forwards compatibility. For example, to get the current active context use the below snippet:
+If a plugin wants to access the context it should use the [provided libraries](../../../runtime/config/context.go) for forwards compatibility. For example, to get the current active context use the below snippet:
```
Comment (anujc25) [inline cli/core/docs/cli/plugin_implementation_guide.md:42]: Thanks for updating this references. 👍

**Review by anujc25 — APPROVED** [review]
Thanks for the updates. LGTM

**Pull request merged**